org.mwc.debrief.core/src/org/mwc/debrief/core/ContextOperations/GeneratePasteRepClipboard.java

```diff
@@ -0,0 +1,215 @@
+          _layers.removeThisLayer((Layer)element);
+        }
+        else {
+          Layer layer = null;
+          if(element instanceof DynamicShapeWrapper) {
+            layer = (Layer)_layers.findLayer(((DynamicShapeWrapper)element).getTrackName());
+          }
+          else if (element instanceof SensorContactWrapper)
+          {
+            layer = (Layer)_layers.findLayer(((SensorContactWrapper)element).getTrackName());
+          }
+          else if (element instanceof DynamicTrackShapeWrapper)
+          {
+            layer = (Layer)_layers.findLayer(((DynamicTrackShapeWrapper)element).getTrackName());
+          }
+          else if (element instanceof TMAContactWrapper)
+          {
+            layer = (Layer)_layers.findLayer(((TMAContactWrapper)element).getTrackName());
+          }
+          else if (element instanceof NarrativeEntry)
+          {
+            layer = _tracker.getLayerFor(ImportReplay.NARRATIVE_LAYER);
+          }
+          else {
+            layer= null;
+          }
+          if(layer!=null) {
+            _layers.removeThisEditable(null, element);
+          }
+        }
+      }
+      _layers.fireExtended();
+      addedElements.clear();
+      return Status.OK_STATUS;
+    }
+  }
+
+  private boolean isContentImportable(final String content) {
+
+    boolean proceed=false;
+    StringTokenizer tokens = new StringTokenizer(content,"\\r?\\n");
+    int lineCount = 0;
+    while(tokens.hasMoreTokens() && lineCount<=6) {
+      String line = tokens.nextToken().trim();
+      if(line.startsWith(";") && !line.startsWith(";;")) {
+        StringTokenizer lineTokens = new StringTokenizer(line);
+        if(lineTokens.hasMoreTokens()) {
+          String firstWord = lineTokens.nextToken();
+          switch(firstWord) {
+            case ";CIRCLE:":
+            case ";DYNAMIC_CIRCLE:":
+            case ";RECT:":
+            case ";DYNAMIC_RECT:":
+            case ";BRG:":
+            case ";DYNAMIC_POLY:":
+            case ";ELLIPSE:":
+            case ";FORMAT_FIX:":
+            case ";FORMAT_LAYER_HIDE:":
+            case ";TEXT:":
+            case ";LINE:":
+            case ";FORMAT_TRACK_NAME_AT_END:":
+            case ";NARRATIVE:":
+            case ";NARRATIVE2:":
+            case ";PERIODTEXT:":
+            case ";POLY:":
+            case ";POLYLINE:":
+            case ";SENSOR:":
+            case ";SENSOR2:":
+            case ";SENSOR3:":
+            case ";SENSOR_ARC:":
+            case ";TIMETEXT:":
+            case ";TMA_POS:":
+            case ";TMA_RB:":
+            case";VECTOR:":
+            case ";WHEEL:":
+              proceed = true;break;
+            default:proceed=false;break;
+          }
+        }
+      }
+      else {
+        StringTokenizer lineTokens = new StringTokenizer(line);
+        if(lineTokens.hasMoreTokens()) {
+          String firstWord = lineTokens.nextToken();
+          if(firstWord.matches("\\d{6}+")) {

```

Comment:
@ayeshare - we do also support an 8 digit date, as described here:
http://www.debrief.info/tutorial/reference.html

Reply:
done